webapp/src/components/markdown_textbox.tsx

```diff
@@ -0,0 +1,221 @@
+// See LICENSE.txt for license information.
+
+import React, {MouseEvent, ChangeEvent, useState} from 'react';
+
+import {useSelector} from 'react-redux';
+
+import classNames from 'classnames';
+import {Link} from 'react-router-dom';
+import styled from 'styled-components';
+
+import {getConfig} from 'mattermost-redux/selectors/entities/general';
+
+import {Textbox} from 'src/webapp_globals';
+
+const DEFAULT_CHAR_LIMIT = 4000;
+
+type MarkdownTextboxProps = {
+    value: string;
+    setValue: (val: string) => void;
+    autocompleteChannelId: string;
+    createMessage?: string;
+    id: string;
+    className?: string
+}
+
+const MarkdownTextbox = ({
+    value,
+    setValue,
+    autocompleteChannelId,
+    className,
+    ...props
+}: MarkdownTextboxProps) => {
+    const [showPreview, setShowPreview] = useState(false);
+    const config = useSelector(getConfig);
+
+    // @ts-expect-error
+    const charLimit = parseInt(config.MaxPostSize || '', 10) || DEFAULT_CHAR_LIMIT;
+
+    return (
+        <div className={className}>
+            <Textbox
+                tabIndex={0}
+                value={value}
+                emojiEnabled={true}
+                supportsCommands={false}
+                suggestionListPosition='bottom'
+                preview={showPreview}
+                useChannelMentions={false}
+                channelId={autocompleteChannelId}
+                onChange={(e: ChangeEvent<HTMLTextAreaElement>) => setValue(e.target.value)}
+                characterLimit={charLimit}
+                createMessage={''}
+                onKeyPress={() => true}
+                openWhenEmpty={true}
+                {...props}
+            />
+            <StyledTextboxLinks
+                characterLimit={charLimit}
+                showPreview={showPreview}
+                updatePreview={setShowPreview}
+                message={value}
+            />
+        </div>
+    );
+};
+
+const StyledMarkdownTextbox = styled(MarkdownTextbox)`
+    .textarea-wrapper {
+        margin-bottom: 6px;
+    }
+
+    &&&& {
+        .form-control.textbox-preview-area {
+            background: rgba(var(--center-channel-color-rgb), 0.04);
+        }
+        .custom-textarea.custom-textarea {
+            height: unset;
+            min-height: 104px;
+            max-height: 324px;
+            overflow: auto;
+            padding: 12px 30px 12px 16px;
+        }
+    }
+`;
+
+type TextboxLinksProps = {
+    showPreview?: boolean;
+    characterLimit: number;
+    updatePreview?: (showPreview: boolean) => void;
+    message: string;
+    className?: string;
+};
+
+function TextboxLinks({
+    message = '',
+    characterLimit,
+    showPreview,
+    className,
+    updatePreview,
+}: TextboxLinksProps) {
+    const togglePreview = (e: MouseEvent) => {
+        e.preventDefault();
+        updatePreview?.(!showPreview);
+    };
+
+    const hasText = message && message.length > 0;
```

Comment:
Can we do this here?

```suggestion
    const hasText = message?.length > 0;
```